src/screens/questionScreen/QuestionForm.js

```diff
@@ -1,296 +1,319 @@
+            questionsData: questionDetail.questions,
+         },
+         Nonmotorist: {
+            reducer: "nonmotoristReducer",
+            submitFunction: updateNonmotorist,
+            actionType: "UPDATENONMOTORIST",
+            questionsData: questionDetail.questions,
+         },
+         Passenger: {
+            reducer: "passengerReducer",
+            submitFunction: updatePassenger,
+            actionType: "UPDATEPASSENGER",
+            questionsData: questionDetail.questions,
+         },
+         Vehicle: {
+            reducer: "vehicleReducer",
+            submitFunction: updateVehicle,
+            actionType: "UPDATEVEHICLE",
+            questionsData: questionDetail.questions,
+         },
+         Road: {
+            reducer: "roadReducer",
+            submitFunction: updateRoad,
+            actionType: "UPDATEROAD",
+            questionsData: questionDetail.questions,
+         },
+      };
+      return type ? obj[type] : obj;
+   };
+
+   /*
+   Returns a question component of the appropriate type with the given props.
+   */
    const renderSingleQuestion = (type, props) => {
       switch (type) {
-        case 'dropdown':
-          return (
-            <DropDownSingleSelect
-              {...props}
-            />
-          )
+         case DROPDOWN:
+            return <DropDownSingleSelect {...props} />;
 
-        case 'dropdownMultiSelect':
+         case DROPDOWN_MULTISELECT:
+            return <DropDownMultiSelect {...props} />;
 
-          return (
-            <DropDownMultiSelect
-              {...props}
-            />
-          )
+         case OPENTEXTBOX:
+            return <OpenTextField {...props} />;
 
-        case 'openTextBox':
+         case OPENTEXTBOX_SELECTION:
+            return <OpenTextFieldWithSelection {...props} />;
 
-          return (
-            <OpenTextField
-              {...props}
-            />
-          )
+         case ADVANCED_OPENTEXTBOX:
+            return <AdvancedOpenTextField {...props} />;
 
-        case 'openTextBoxWithSelection':
+         case ADVANCED_DROPDOWN:
+            return <AdvancedDropDown {...props} />;
 
-          return (
-            <OpenTextFieldWithSelection
-              {...props}
-            />
-          )
+         case COUNTY_DROPDOWN:
+            return <CountyDropDown {...props} />;
 
-          case 'advancedOpenTextBox':
+         case LARGE_TEXT_FIELD:
+            return <LargeTextField {...props} />;
 
-            return (
-              <AdvancedOpenTextField
-              {...props}
-              />
-            )
+         case MULTIBUTTON:
+            return <MultiButtonSelector {...props} />;
 
-        case 'advancedDropDown':
+         case NUMBERBUTTON:
+            return <NumberButtonSelector {...props} />;
 
-          return (
-            <AdvancedDropDown
-              {...props}
-            />
-          )
+         case AUTOCOMPLETE_DROPDOWN:
```

Comment:
Is this being used anywhere?

Reply:
Good question! I don't think we are actually using this component anywhere. None of the questions are configured to have an autocomplete dropdown. Should I delete the component?

Reply:
Yes